File: MovieManagement_BE/MovieManagement/Data/Entities/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace MovieManagement.Data.Entities;

public class Movie
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Genre { get; set; }

    [Range(1, 5)]
    public int? Rating { get; set; }

    [MaxLength(500)]
    public string? PosterImage { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }
}

File: MovieManagement_FE/movie-management/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  genre?: string;
  rating?: number;
  posterImage?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateMovieDto {
  title: string;
  genre?: string;
  rating?: number;
  posterImage?: string;
}

export interface UpdateMovieDto {
  title: string;
  genre?: string;
  rating?: number;
  posterImage?: string;
}